lib/jxl/dec_patch_dictionary.cc

```diff
@@ -177,44 +177,138 @@ int PatchDictionary::GetReferences() const {
   return result;
 }
 
-void PatchDictionary::ComputePatchCache() {
-  patch_starts_.clear();
-  sorted_patches_.clear();
-  if (positions_.empty()) return;
-  std::vector<std::pair<size_t, size_t>> sorted_patches_y;
-  for (size_t i = 0; i < positions_.size(); i++) {
-    const PatchPosition& pos = positions_[i];
-    const PatchReferencePosition& ref_pos = ref_positions_[pos.ref_pos_idx];
-    for (size_t y = pos.y; y < pos.y + ref_pos.ysize; y++) {
-      sorted_patches_y.emplace_back(y, i);
-    }
+namespace {
+struct PatchInterval {
+  size_t idx;
+  size_t y0, y1;
+};
+}  // namespace
+
+void PatchDictionary::ComputePatchTree() {
+  patch_tree_.clear();
+  num_patches_.clear();
+  sorted_patches_y0_.clear();
+  sorted_patches_y1_.clear();
+  if (positions_.empty()) {
+    return;
   }
-  // The relative order of patches that affect the same pixels is preserved.
```

Comment:
Is this still true?

Reply:
Not sure, probably not, should we add a sort at the end of GetPatchesForRow then?

Reply:
Probably yes, as patch application is not necessarily commutative :)

Reply:
Done.